### ISSUE TITLE
EdvDocument does not use the keyAgreementKey from the `client` unless it is passed in explicitly

### ISSUE THREAD

**Issue by mattcollier** [digitalbazaar/edv-client#45]
```
          const edvDocument = new EdvDocument({
            capability: currProfileAgent.zcaps.capabilitySetDocument,
            client: capabilitySetEdv,
            // FIXME: EdvDocument does not use the keyAgreementKey from the
            // client unless it is passed in explicitly
            keyAgreementKey: capabilitySetEdv.keyAgreementKey,
            invocationSigner: invocationSigner.invocationSigner,
          });
```

https://github.com/digitalbazaar/edv-client/blob/master/EdvDocument.js#L61

An `undefined` `keyAgreementKey` parameter will override the keyAgreementKey that is already included in the client that is passed in.  

It appears to me that `keyAgreementKey` and `client` are mutually exclusive options for this reason.  Or perhaps the `keyAgreementKey` parameter should *only* be coming through a `client`?

**Comment by dlongley** [comment]
I think this is just a mistake with the defaults in the EdvDocument constructor. Several are set to `null` and they should be removed so that the parameters default to `undefined` instead.

**Comment by mattcollier** [comment]
@dlongley I see that, that would be a simple fix, however, what is the use case for passing in *both* a client and a keyAgreementKey?  Isn't it expected that an edvClient instance will always have a keyAgreementKey internally?

**Comment by dlongley** [comment]
>  I see that, that would be a simple fix, however, what is the use case for passing in both a client and a keyAgreementKey?

The client could have a different `keyAgreementKey` (or none at all) associated with it.

> Isn't it expected that an edvClient instance will always have a keyAgreementKey internally?

No, I don't think so.

**Comment by mattcollier** [comment]
Fix landed in 2.5.2.
